infra/cifuzz/cifuzz_test.py

```diff
@@ -217,5 +218,104 @@ def test_parse_invalid_output(self):
+    self.assertCountEqual(file_list, true_files_list)
+
+  def test_invalid_target(self):
+    """Test asserts an invalid fuzzer returns None."""
+    self.assertIsNone(
+        cifuzz.get_files_covered_by_target(self.envoy_cov_exmp, 'not-a-fuzzer',
+                                           '/src/curl'))
+    self.assertIsNone(
+        cifuzz.get_files_covered_by_target(self.envoy_cov_exmp, '',
+                                           '/src/curl'))
+
+  def test_invalid_project_build_dir(self):
+    """Test asserts an invalid build dir returns None."""
+    self.assertIsNone(
+        cifuzz.get_files_covered_by_target(self.envoy_cov_exmp,
+                                           self.example_fuzzer, '/no/pe'))
+    self.assertIsNone(
+        cifuzz.get_files_covered_by_target(self.envoy_cov_exmp,
+                                           self.example_fuzzer, ''))
+
+
+class GetTargetCoverageReportIntegrationTest(unittest.TestCase):
+  """Test get_target_coverage_report function in the cifuzz module."""
+
+  example_cov_json = 'example_curl_cov.json'
+  example_fuzzer = 'curl_fuzzer'
+
+  def setUp(self):
+    with open(os.path.join(TEST_FILES_PATH, self.example_cov_json),
+              'r') as file:
+      self.cov_exmp = json.loads(file.read())
+
+  def test_valid_target(self):
+    """Test a target's coverage report can be downloaded and parsed."""
+    target_report = cifuzz.get_target_coverage_report(self.cov_exmp,
+                                                      self.example_fuzzer)
+    self.assertIsNotNone(target_report['data'][0]['files'])
+    self.assertIsNotNone(target_report['version'])
+    self.assertEqual(840, len(target_report['data'][0]['files']))
+
+  def test_invalid_target(self):
+    """Test an invalid target coverage report will be None."""
+    self.assertIsNone(
+        cifuzz.get_target_coverage_report(self.cov_exmp, 'not-valid-target'))
+    self.assertIsNone(cifuzz.get_target_coverage_report(self.cov_exmp, ''))
+
+  def test_invalid_project_json(self):
+    """Test a project json coverage report will be None."""
+    self.assertIsNone(
+        cifuzz.get_target_coverage_report('not-a-proj', self.example_fuzzer))
+    self.assertIsNone(cifuzz.get_target_coverage_report('',
+                                                        self.example_fuzzer))
+
+
+class GetProjectCoverageReportIntegrationTest(unittest.TestCase):
+  """Test get_project_coverage_report function in the cifuzz module."""
+
+  test_project = 'curl'
+
+  def test_get_valid_project(self):
+    """Tests that a project's coverage report can be downloaded and parsed.
+
+    NOTE: This test relies on the test_project repo's coverage report.
+    Example was not used because it has no coverage reports.
+    """
+    cov_report = cifuzz.get_project_coverage_report(self.test_project)
+    self.assertIsNotNone(cov_report)
+    self.assertEqual(type(cov_report), dict)
+    self.assertTrue('fuzzer_stats_dir' in cov_report)
```

Comment:
I don't think you need to test such details like these keys, just mock the download method and assert that it is being called with the correct expected URL

Reply:
Fixed.